src/remount.rs

```diff
@@ -0,0 +1,439 @@
+            try!(write!(fmt, "{}nodev", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.noexec {
+            try!(write!(fmt, "{}noexec", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.nosuid {
+            try!(write!(fmt, "{}nosuid", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.noatime {
+            try!(write!(fmt, "{}noatime", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.nodiratime {
+            try!(write!(fmt, "{}nodiratime", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.relatime {
+            try!(write!(fmt, "{}relatime", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.strictatime {
+            try!(write!(fmt, "{}strictatime", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.dirsync {
+            try!(write!(fmt, "{}dirsync", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.synchronous {
+            try!(write!(fmt, "{}sync", prefix));
+            prefix = ",";
+        }
+        if let Some(true) = self.mandlock {
+            try!(write!(fmt, "{}mand", prefix));
+        }
+        Ok(())
+    }
+}
+
+impl fmt::Display for Remount {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        if !self.flags.apply_to_flags(MsFlags::empty()).is_empty() {
+            try!(write!(fmt, "{} ", self.flags));
+        }
+        write!(fmt, "remount {:?}", &self.path)
+    }
+}
+
+impl Explainable for Remount {
+    fn explain(&self) -> String {
+        [
+            format!("path: {}", exists(&self.path)),
+            format!("{}", user()),
+        ].join(", ")
+    }
+}
+
+fn get_mountpoint_flags(path: &Path) -> Result<MsFlags, RemountError> {
+    let mount_path = if path.is_absolute() {
+        path.to_path_buf()
+    } else {
+        let mut mpath = try!(current_dir());
+        mpath.push(path);
+        mpath
+    };
+    let mut mountinfo_content = vec!();
+    let mountinfo_path = Path::new("/proc/self/mountinfo");
+    let mut mountinfo_file = try!(File::open(mountinfo_path)
+        .map_err(|e| RemountError::Io(
+            format!("Cannot open file: {:?}", mountinfo_path), e)));
+    try!(mountinfo_file.read_to_end(&mut mountinfo_content)
+        .map_err(|e| RemountError::Io(
+            format!("Cannot read file: {:?}", mountinfo_path), e)));
+    match get_mountpoint_flags_from(&mountinfo_content, &mount_path) {
+        Ok(Some(flags)) => Ok(MsFlags::from_bits_truncate(flags)),
+        Ok(None) => Err(RemountError::UnknownMountPoint(mount_path)),
+        Err(e) => Err(e),
+    }
+}
+
+fn get_mountpoint_flags_from(content: &[u8], path: &Path)
+    -> Result<Option<c_ulong>, RemountError>
+{
+    let mounts_parser = MountInfoParser::new(content);
+    let mut flags = None;
+    for mount_info_res in mounts_parser {
```

Comment:
Additional thought: You should use reverse iterator and break as soon as possible.